modules/fileout_netcdf/FONcArray.cc

```diff
@@ -217,6 +242,125 @@ void FONcArray::convert(vector<string> embed, bool _dap4, bool is_dap4_group) {
+    // netcdf file. So reduce the chunk size when the similar case occurs.
+    // The following rules handle the cases described above.
+    // Given there may be a very large size array and also the chunk size cannot too big,
+    // we modify the maximum chunk size according to the array size if necessary. 
+    // The idea is we don't want to have too many chunks and we don't want to have the chunk size too big.
+    // So we do the following. 
+    // 1. We start from 1M(1024x1024), if the number of chunks for the array is >64, we increase the chunk size to be 2048x2048.
+    // 2. We can increase the chunk size to 16M(4096x4096) if the number of chunks is still >64.
+    // 3. The maximum chunk size is 16M(4096x4096) no matter how big the array size is.
+    // 4. We can increase the maximum chunk size to a bigger number in the future if necessary.
+    //    We can also change the maximum number of chunks.
+    // KY 2023-01-26
+
+    // The above algorithm works well to reduce the number of chunks for a large array size as well as having a decent
+    // access time. However,
+    // it may increase the final file size when the fastest changing dimension size increases from 1024 to 2048 or 4096.
+    // So to find a good balance between the file size and the access time for NASA files,
+    // we fix the sizes of two fastest changing dimensions to be <=1024x1024. We will increase the higher chunk dimension size 
+    // from 1 to 2 or higher if the number of higher dimension size is >512. Why 512? Because we find the higher dimension
+    // may be time and one year is 365 days. We want to still keep the one day per slice if possible. 512 is not too big. 
+    // The chunk overhead is not that big.
+    // KY 2023-01-29
+    // One more optimiziation: if the sizes of two fastest changing dimensions are much less than 1M(1024x1024),given
+    // that the higher dimension size may be large, we allow the chunk size grow to 1M and then if we still find the number of
+    // higher dimension size is >512, we then grow the chunk size gradually. This will make the bigger array size not hold
+    // too many chunks.
+
+    // Chunk size is not set when the number of dimension is >2. It is set here.
+    // When the number of higher than 2 dimension sizes is <max_num_higher_chunks(512 now), we start increasing the higher chunk dimension size. 
+    // KY 2023-01-29
+    if (d_a->dimensions() >2) {
+
+        // The chunk sizes of the two fastest changing dimensions are fixed(not exceeding the MAX_CHUNK_SIZE). So set them first.
+        size_t two_fastest_chunk_dim_sizes=1;
+        auto d_chunksize_it = d_chunksizes.begin();
+
+        for (size_t i = d_a->dimensions();i>d_a->dimensions()-2; i--) {
+
+            size_t size = d_dim_sizes[i-1];
+            BESDEBUG("fonc", "FONcArray::CHUNK - dim size backward: " << size << endl);
+            d_chunksize_it = d_chunksizes.insert(d_chunksize_it, size <= MAX_CHUNK_SIZE ? size : MAX_CHUNK_SIZE);
```

Comment:
For things like "size <= MAX_CHUNK_SIZE ? size : MAX_CHUNK_SIZE" you could use "min(size, MAX_CHUNK_SIZE)" or std::min(...) although I'm not sure there's a huge difference in performance. I think min() & max() are a bit easier to read, but it's probably not worth too much effort.